sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/CombineFns.java

```diff
@@ -470,7 +589,10 @@ public CoCombineResult extractOutput(Object[] accumulator, Context c) {
         throws CannotProvideCoderException {
       List<Coder<Object>> coders = Lists.newArrayList();
       for (int i = 0; i < combineFnCount; ++i) {
-        Coder<Object> inputCoder = registry.getOutputCoder(extractInputFns.get(i), dataCoder);
+        Coder<Object> inputCoder =
+            combineInputCoders.get(i).isPresent()
```

Comment:
Unfortunately the get.or method won't work here, since it's type signature can't handle functions that throw exceptions (such as CannotProvideCoderExceptions).

As to the second, I don't think the registry is available at that time the combiner is built up. The registry is generally not available until later in pipeline construction, and the Combiner signature doesn't give it to us until getAccumulatorCoder is called.